File: solving-problems/circular-array-rotation.py
def circularArrayRotation(a, k, queries):

    n = len(a)
    arr_aux = []

    for i in range(k):
        a.insert(0,a[n-1])
        a.pop()
        
    for i in range(len(queries)):
        arr_aux.append(a[queries[i]])
    return arr_aux

File: solving-problems/circular-array-rotation.java
public static List<Integer> circularArrayRotation(List<Integer> a, int k, List<Integer> queries) {
    
    int n = a.size();
    List<Integer> arr_aux = new ArrayList<>();

    for(int i=0; i<k; i++){
        a.add(0,a.get(n-1));
        a.remove(n);
    }
            
    for(int i=0; i<queries.size(); i++){
        arr_aux.add(a.get(queries.get(i)));
    }
    return arr_aux;

}